plugins/spi/SPIOutputTest.cpp

```diff
@@ -421,3 +425,277 @@ void SPIOutputTest::testCombinedP9813Control() {
+
+  // check for Equality
+  OLA_ASSERT_DATA_EQUALS(EXPECTED0, arraysize(EXPECTED0), data, length);
+  // check if the output writes are 1
+  OLA_ASSERT_EQ(1u, backend.Writes(0));
+
+
+
+  // test2
+  buffer.SetFromString("255,128,0,10,20,30");
+  output.WriteDMX(buffer);
+  data = backend.GetData(0, &length);
+  const uint8_t EXPECTED1[] = { 0, 0, 0, 0,
+                                0xff, 0, 0x80, 0xff,
+                                0xff, 0x1e, 0x14, 0x0a,
+                                0};
+  OLA_ASSERT_DATA_EQUALS(EXPECTED1, arraysize(EXPECTED1), data, length);
+  OLA_ASSERT_EQ(2u, backend.Writes(0));
+
+  // test3
+  // test what happens when only new data for the first leds is available.
+  // later data should be not modified so data set in test2 is valid for pixel2
+  buffer.SetFromString("34,56,78");
+  output.WriteDMX(buffer);
+  data = backend.GetData(0, &length);
+  // print data to see what is going on.
+  // OLA_INFO << "length: " << length;
+  // OLA_INFO << "data: ";
+  // for (unsigned int i = 0; i < length; i++) {
+    // OLA_INFO << "[" << i << "] " <<  static_cast<int>(data[i]);
+  // }
+  const uint8_t EXPECTED2[] = { 0, 0, 0, 0,
+                                0xff, 0x4e, 0x38, 0x22,
+                                0xff, 0x1e, 0x14, 0x0a,
+                                0};
+  OLA_ASSERT_DATA_EQUALS(EXPECTED2, arraysize(EXPECTED2), data, length);
+  OLA_ASSERT_EQ(3u, backend.Writes(0));
+
+  // test4
+  // tests what happens if fewer then needed color information are received
+  buffer.SetFromString("7, 9");
+  output.WriteDMX(buffer);
+  data = backend.GetData(0, &length);
+  // check that the returns are the same as test3 (nothing changed)
+  OLA_ASSERT_DATA_EQUALS(EXPECTED2, arraysize(EXPECTED2), data, length);
+  OLA_ASSERT_EQ(3u, backend.Writes(0));
+
+  // test5
+  // test with changed StartAddress
+  // set StartAddress
+  output.SetStartAddress(3);
+  // values 1 & 2 should not be visible in SPI data stream
+  buffer.SetFromString("1,2,3,4,5,6,7,8");
+  output.WriteDMX(buffer);
+  data = backend.GetData(0, &length);
+  const uint8_t EXPECTED4[] = { 0, 0, 0, 0,
+                                0xff, 0x05, 0x04, 0x03,
+                                0xff, 0x08, 0x07, 0x06,
+                                0};
+  OLA_ASSERT_DATA_EQUALS(EXPECTED4, arraysize(EXPECTED4), data, length);
+  OLA_ASSERT_EQ(4u, backend.Writes(0));
+  // change StartAddress back to default
+  output.SetStartAddress(1);
+
+  // test6
+  // Check nothing changed on the other output.
+  OLA_ASSERT_EQ(reinterpret_cast<const uint8_t*>(NULL),
+                backend.GetData(1, &length));
+  OLA_ASSERT_EQ(0u, backend.Writes(1));
+
+  /**
+  // test7
+  // create new output with pixel_count=16 and check data length
+  // setup pixel_count to 16
+  options.pixel_count = 16;
+  // setup SPIOutput
+  SPIOutput output(m_uid, &backend, options);
+  // set personality
+  output.SetPersonality(thisTestPersonality);
```

Comment:
At this point i tried to add two tests to test the endFrame length.
but as fare as i found it is not possible to change the pixel_count after the initialization of the output.
and i did not know how to 'reinitialize' the output with new options..